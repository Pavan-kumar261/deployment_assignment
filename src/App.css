.radio_buttons{
    text-align: center;
}
tr{
    margin-top: 10px;
}

td{
    padding-left:10px
}