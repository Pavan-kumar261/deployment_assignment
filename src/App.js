import React from "react";
import { Trans, withTranslation } from "react-i18next";
import "./App.css";
import "./i18n.js";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: "en",
    };
  }

  onLanguageHandle = (event) => {
    let newLang = event.target.value;
    this.setState({ value: newLang });
    this.props.i18n.changeLanguage(newLang);
  };

  renderRadioButtons = () => {
    return (
      <div>
        <input
          checked={this.state.value === "en"}
          name="language"
          onChange={(e) => this.onLanguageHandle(e)}
          value="en"
          type="radio"
        />
        English &nbsp;
        <input
          name="language"
          value="ru"
          checked={this.state.value === "ru"}
          type="radio"
          onChange={(e) => this.onLanguageHandle(e)}
        />
        Russian &nbsp;
        <input
          name="language"
          value="jp"
          checked={this.state.value === "jp"}
          type="radio"
          onChange={(e) => this.onLanguageHandle(e)}
        />
        Japanese &nbsp;
        <input
          name="language"
          value="hi"
          checked={this.state.value === "hi"}
          type="radio"
          onChange={(e) => this.onLanguageHandle(e)}
        />
        Hindi &nbsp;
        <input
          name="language"
          value="gk"
          checked={this.state.value === "gk"}
          type="radio"
          onChange={(e) => this.onLanguageHandle(e)}
        />
        Greek
      </div>
    );
  };

  render() {
    const { t } = this.props;
    return (
      <div className="App">
        <h1 style={{ textAlign: "center" }}>Translate App</h1>
        <p className="radio_buttons">{this.renderRadioButtons()}</p>
        <h1>
          <Trans>{t("paragraph")}</Trans>
        </h1>
        <table>
          <tbody>
            <tr>
              <td>{t("author.title")}</td>
              <td>:</td>
              <td>{t("author.value")}</td>
            </tr>
            <tr>
              <td>{t("description.title")}</td>
              <td>:</td>
              <td>{t("description.value")}</td>
            </tr>
          </tbody>
        </table>

       
        <div style={{ marginTop: 40 }}>
          Learn more:&nbsp;
          <a href="https://react.i18next.com">https://react.i18next.js</a>
        </div>
      </div>
    );
  }
}

export default withTranslation()(App);
