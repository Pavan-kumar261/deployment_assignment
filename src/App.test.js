import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import App from "./App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Matches the snapshot", () => {
  const component = TestRenderer.create(<App />);
  expect(component).toMatchSnapshot();
});
