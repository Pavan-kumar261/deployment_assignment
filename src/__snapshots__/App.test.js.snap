// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Translate App
  </h1>
  <p
    className="radio_buttons"
  >
    <div>
      <input
        checked={true}
        name="language"
        onChange={[Function]}
        type="radio"
        value="en"
      />
      English  
      <input
        checked={false}
        name="language"
        onChange={[Function]}
        type="radio"
        value="ru"
      />
      Russian  
      <input
        checked={false}
        name="language"
        onChange={[Function]}
        type="radio"
        value="jp"
      />
      Japanese  
      <input
        checked={false}
        name="language"
        onChange={[Function]}
        type="radio"
        value="hi"
      />
      Hindi  
      <input
        checked={false}
        name="language"
        onChange={[Function]}
        type="radio"
        value="gk"
      />
      Greek
    </div>
  </p>
  <h1>
    Quotes
  </h1>
  <table>
    <tbody>
      <tr>
        <td>
          Author Name
        </td>
        <td>
          :
        </td>
        <td>
          E. L. Doctorow
        </td>
      </tr>
      <tr>
        <td>
          Quote
        </td>
        <td>
          :
        </td>
        <td>
          Writing a novel is like driving a car at night. You can only see as far as your headlights, but you can make the whole trip that way
        </td>
      </tr>
    </tbody>
  </table>
  <div
    style={
      Object {
        "marginTop": 40,
      }
    }
  >
    Learn more: 
    <a
      href="https://react.i18next.com"
    >
      https://react.i18next.js
    </a>
  </div>
</div>
`;
